lightrag/operate.py

```diff
@@ -399,42 +412,63 @@ async def local_query(
     global_config: dict,
 ) -> str:
     context = None
-    use_model_func = global_config["llm_model_func"]
+    example_number = global_config["addon_params"].get("example_number", None)
+    if example_number and example_number < len(PROMPTS["keywords_extraction_examples"]):
+        examples = "\n".join(
+            PROMPTS["keywords_extraction_examples"][: int(example_number)]
+        )
+    else:
+        examples = "\n".join(PROMPTS["keywords_extraction_examples"])
+
+    # Set mode
+    if query_param.mode not in ["local", "global", "hybrid"]:
+        logger.error(f"Unknown mode {query_param.mode} in kg_query")
+        return PROMPTS["fail_response"]
 
+    # LLM generate keywords
+    use_model_func = global_config["llm_model_func"]
     kw_prompt_temp = PROMPTS["keywords_extraction"]
-    kw_prompt = kw_prompt_temp.format(query=query)
+    kw_prompt = kw_prompt_temp.format(query=query, examples=examples)
     result = await use_model_func(kw_prompt)
-    json_text = locate_json_string_body_from_string(result)
-
+    logger.info("kw_prompt result:")
+    print(result)
     try:
+        json_text = locate_json_string_body_from_string(result)
         keywords_data = json.loads(json_text)
-        keywords = keywords_data.get("low_level_keywords", [])
-        keywords = ", ".join(keywords)
-    except json.JSONDecodeError:
-        try:
-            result = (
```

Comment:
This section is an adaptation for responses that don't fully adhere to the JSON format. It needs to be kept as is. For example, some responses are enclosed with:
\```json
xxxxxxxxx
\```

Reply:
The function `locate_json_string_body_from_string` use `re` to parse json text, all formate like`{......}` can be  matched. So I think no need to prcess the LLM output text use `replace`.